eng/common/scripts/TypeSpec-Project-Generate.ps1

```diff
@@ -14,6 +14,78 @@ $ErrorActionPreference = "Stop"
 . $PSScriptRoot/common.ps1
 Install-ModuleIfNotInstalled "powershell-yaml" "0.4.1" | Import-Module
 
+function GetNpmPackageVersion([string]$packagePath) {
+    $packagePath = Resolve-Path $packagePath
+    $packageFolder = Split-Path -Parent $packagePath
+    $packageName = Split-Path -Leaf $packagePath
+    Push-Location $packageFolder
+    try {
+        $packageVersion = node -p -e "require('./$packageName').version"
+        return $packageVersion
+    }
+    finally {
+        Pop-Location
+    }
+}
+
+function GetEmitterDependencyVersion([string]$packagePath) {
+    $emitterName = &$GetEmitterNameFn
+    $matchString = "`"$emitterName`": `"(.*?)`""
+    If ((Get-Content -Raw $packagePath) -match $matchString) {
+        return $Matches[1]
+    }
+}
+
+function NpmInstallAtRoot() {
+    $root = git rev-parse --show-toplevel
+    Push-Location $root
+    try {
+        #default to root/eng/emitter-package.json but you can override by writing
+        #Get-${Language}-EmitterPackageJsonPath in your Language-Settings.ps1
+        $replacementPackageJson = "$PSScriptRoot/../../emitter-package.json"
```

Comment:
We should probably pick either `$root` or `$PSScriptRoot` and be consistent with that throughout.